core/src/main/java/com/squarespace/template/plugins/platform/ContentFormatters.java

```diff
@@ -502,13 +503,30 @@ protected void outputImageSrcSet(JsonNode image, StringBuilder buf) {
         }
         buf.append(assetUrl).append("?format=").append(variants[i]).append(" ").append(variants[i]);
       }
-
-      buf.append(',');
-      buf.append(assetUrl).append("?format=original");
```

Comment:
Could you just inline the logic and scan the sizes backwards, since most images should have the larger resolutions available?  Something like:

```
String lastVariant = variants[limit - 1];
int sizeLimit = SQUARESPACE_SIZES.length - 1;
for (int i = sizeLimit; i >= 0; i--) {
  // If the largest variant is not the largest available resolution..
  if (lastVariant.equals(SQUARESPACE_SIZES[i]) && i < sizeLimit) {
    // Append the original image as the next size up
    buf.append(',');
    buf.append(assetUrl).append("?format=original").append(' ').append(SQUARESPACE_SIZES[i + 1]);
    break;
  }
}
```

Reply:
inlined the logic and scanning the array backwards. thanks